nuxeo-core/nuxeo-core-storage-sql/nuxeo-core-storage-sql/src/main/java/org/nuxeo/ecm/core/storage/sql/jdbc/JDBCRowMapper.java

```diff
@@ -1077,8 +1077,8 @@ protected Serializable copyHierRecursive(IdWithTypes source, Serializable parent
         }
         // recurse in children
         boolean onlyComplex = parentId == null;
-        for (IdWithTypes child : getChildrenIdsWithTypes(source.id, onlyComplex)) {
-            copyHierRecursive(child, newId, null, null, resetVersion, idMap, idToTypes);
+        for (IdWithTypes child : getChildrenIdsWithTypes(source.id, onlyComplex, excludeUnwantedChildren)) {
+            copyHierRecursive(child, newId, null, null, resetVersion, idMap, idToTypes, excludeUnwantedChildren);
```

Comment:
I thought that during recursion we didn't want to excluded special children?

Reply:
If I checkin , I want not to exclude special children like CommentRoot so I can copy it.
But I also want to copy the regular children under CommentRoot so I need to reverse the flag and keep it at true. 
So the call to the recursion has a hardcoded flag that allows me to copy all the comments under the CommentRoot